src/main/java/org/openpnp/machine/reference/vision/ReferenceFiducialLocator.java

```diff
@@ -177,40 +183,49 @@ public Location getHomeFiducialLocation(Location location, Part part) throws Exc
      * @return
      * @throws Exception
      */
-    private static Location getFiducialLocation(BoardLocation boardLocation, Placement fid)
-            throws Exception {
+	private static Location getFiducialLocation(BoardLocation boardLocation, Placement fid, Length fidDia) throws Exception {
         Camera camera = Configuration.get().getMachine().getDefaultHead().getDefaultCamera();
 
         Logger.debug("Locating {}", fid.getId());
 
+		Footprint footprint;
+
+		// Careful...magic number below. Also used in JobPanel. This are for
+		// global panel fiducials, which do not have an entry in the PCB BOM.
+		if ((fid.getId().equals("PanelFid1_a322") || fid.getId().equals("PanelFid2_g301"))) {
```

Comment:
Hi @BendRocks, thanks for your detailed response. I agree with most of your points. Here is my current thinking on this:

Biggest issue is the fiducial handling:
* Current code only supports round fids. Certainly the most common, but it's not rare to have others.
* An in-process PR adds pipeline configuration for fids, allowing vision customization. Happens at Part level, like bottom vision. Skipping Part/Package for panel fids will make this feature not work for panels.
* The hardcoded placement IDs are a brittle way to handle the problem. Gut feeling is that we should just pass the BoardLocation like we do now and if the fid locator sees that the BoardLocation is a panel it uses panel fids instead of board fids.
* Fids are not checked during job startup, and there's no way to indicate that they should be. This could be considered an enhancement down the road - not required for v1.

Here is roughly how I'd like to see this feature change in the code:
* Panel class (rename of PCBPanel):
    * Contains panel info such as columns, rows, xGap, yGap.
    * Adds IdentifiableList<Placement> placements for panel fids. Can also be used in future for non-board placements, if needed.
    * Adds "Check Fids?" checkbox which indicates the panel fids should be checked at job start.
* Job gets IdentifiableList<Panel> panels.
* BoardLocation gets panelId which refers to a Panel in Job.panels. We'll also materialize a BoardLocation.panel property during job load so that BoardLocation references can get you back to the Panel. This will be used in the fid checker.

And here is how it would all work:
* Dialog and buttons all remain basically the same as they are now.
* Dialog can look up the panel info based on a BoardLocation selection via BoardLocation.panel.
* Remove the fid diameter and add fid part dropdown. The dialog saves two placements into the Panel.placements with the specified coordinates and part.
* FiducialLocator API can just get a boolean checkPanel which tells it to do a panel check rather than an individual board check. It uses BoardLocation.panel to find the panel fids and then everything else works basically the same.

I think this will make the feature much more flexible and allow it to be grown in the future as needed. It will also make it compatible with the upcoming fiducial vision changes, which is important. And I don't actually think it is a very large set of changes from what you have now.

I know you said you have some other pressing work at the moment, so if this is too much to get to right away we can either sit on it for a while or I can work on it. I think I can probably refactor it into the above in a night or two and I'd be happy to do that to get the feature merged in before it gets too stale.

Let me know what you think. This is just my thinking on it and it's not a dictation. Completely open to discussion.